service/history/stateBuilder.go

```diff
@@ -350,39 +349,44 @@ func (b *stateBuilderImpl) applyEvents(domainID, requestID string, execution sha
 				WorkflowId: execution.WorkflowId,
 				RunId:      common.StringPtr(newRunID),
 			}
-
 			// Create mutable state updates for the new run
 			newRunStateBuilder = newMutableStateBuilderWithReplicationState(b.shard.GetConfig(), b.logger, startedEvent.GetVersion())
 			newRunStateBuilder.ReplicateWorkflowExecutionStartedEvent(domainID, parentDomainID, newExecution, uuid.New(),
 				startedAttributes)
-			di := newRunStateBuilder.ReplicateDecisionTaskScheduledEvent(
-				dtScheduledEvent.GetVersion(),
-				dtScheduledEvent.GetEventId(),
-				dtScheduledEvent.DecisionTaskScheduledEventAttributes.TaskList.GetName(),
-				dtScheduledEvent.DecisionTaskScheduledEventAttributes.GetStartToCloseTimeoutSeconds(),
-			)
+
+			var di *decisionInfo
+			nextEventID := startedEvent.GetEventId() + 1
+			if len(newRunHistory.Events) > 1 {
+				dtScheduledEvent := newRunHistory.Events[1]
+				di = newRunStateBuilder.ReplicateDecisionTaskScheduledEvent(
+					dtScheduledEvent.GetVersion(),
+					dtScheduledEvent.GetEventId(),
+					dtScheduledEvent.DecisionTaskScheduledEventAttributes.TaskList.GetName(),
+					dtScheduledEvent.DecisionTaskScheduledEventAttributes.GetStartToCloseTimeoutSeconds(),
+				)
+				nextEventID = di.ScheduleID + 1
+			}
 			newRunExecutionInfo := newRunStateBuilder.GetExecutionInfo()
-			nextEventID := di.ScheduleID + 1
 			newRunExecutionInfo.NextEventID = nextEventID
 			newRunExecutionInfo.LastFirstEventID = startedEvent.GetEventId()
 			// Set the history from replication task on the newStateBuilder
 			newRunStateBuilder.SetHistoryBuilder(newHistoryBuilderFromEvents(newRunHistory.Events, b.logger))
 			sourceClusterName := b.clusterMetadata.ClusterNameForFailoverVersion(startedEvent.GetVersion())
 			newRunStateBuilder.UpdateReplicationStateLastEventID(sourceClusterName, startedEvent.GetVersion(), nextEventID-1)
 
-			b.newRunTransferTasks = append(b.newRunTransferTasks, b.scheduleDecisionTransferTask(domainID,
-				b.getTaskList(newRunStateBuilder), di.ScheduleID))
+			if startedAttributes.GetAttempt() == 0 {
```

Comment:
if attempt is !=0, then no decision is scheduled?
this can be fine if the workflow is being replicated, however, when doing a failover and the new attempt workflow is picked up by the previous standby cluster (now active), the no worker can pick this workflow up. 
this workflow will be invisible until the decision task scheduled event is replicated from active cluster (now standby after the failover). however, there is no guarantee that this decision task scheduled event will be replicated.

Reply:
the first scheduled event will be created by a backoff timer, the timer should fire and because it is now become active, it would create the scheduled event locally.

Reply:
I think the stand by timer queue process will need to be updated to handle that case. will update it.

Reply:
@yiminc the standby processor only does verification, as long as the timer task is created, the failover will handle the task processing

Reply:
the failover processing logic is the same as the active processing logic